### DISCUSSION TITLE
Fetch shared presets from HTTP URLs

### DISCUSSION THREAD

**Discussion by amezin** [renovatebot/renovate#27309] [Suggest an Idea]
### Tell us more.

I want to use shared presets, but all repositories are hosted on bitbucket server (not cloud).

Also, I want to still be able to run Renovate with `--platform=local`, to test config changes (mostly custom manager regexes) without pushing/merging them to the default branch.

Would it be possible to allow simple HTTP URLs in "extends" list? To me it looks like an obvious (and trivial to implement) idea, are there any obstacles I'm overlooking?

**Comment by rarkins** [comment]
Could you give an example of the syntax you have in mind?

**Reply by amezin** [reply]
```json
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [
        "config:recommended",
        "https://our.bitbucket.server/users/amezin/repos/renovate-presets/raw/default.json?at=refs%2Fheads%2Fmaster"
    ]
}
```

`http://` / `https://` prefix can be an indicator that it's an URL, and not the current syntax

**Reply by rarkins** [reply]
This could possibly work. So it would need to be an endpoint which returns JSON?

**Reply by amezin** [reply]
Yes, or JSON5. The same format(s) as when fetching from other sources (`github>`, `gitlab>`, `local>`)
